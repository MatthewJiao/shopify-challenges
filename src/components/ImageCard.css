.cardBorderStyle {
  background: url(../assets/back7.png);
}
