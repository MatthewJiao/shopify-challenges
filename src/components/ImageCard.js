import React, {useState} from 'react'
import parse from 'de-noun-parser';


const ImageCard = ({image}) => {
  const tags = image.tags.replaceAll(',', '').split(' ')
  console.log(tags)
  
  
    return (
      <div style = {cardBorderStyle} className = "max-w-sm rounded-md overflow-hidden shadow-lg">
        <img src = {image.webformatURL} alt = {image.largeImageURL} style = {imageStyle} className = "w-full"/>
          <div className = "px-6 py-4">
           <div style = {titleStyle} className = "font-bold text-xl mb-2">
            
        </div>
        <ul style = {infoStyle}>
          <li style = {{color: "#31948e"}}>
            <strong>View:</strong> {image.views}
          </li>
          <li>
            <strong>Downloads:</strong> {image.downloads}
          </li>
          <li>
            <strong>Likes:</strong> {image.likes}
          </li>
          
        </ul>
      </div>

      <div style = {botStyle} className = "px-6 py-4 pt-1">
        <div>
          {tags.map((tag, index) => (
              <span style = {tagStyle} key = {index} className = "inline-block rounded-md px-3 py-1 text-sm font-semibold text-gray-700 mr-2">
              #{tag}
            </span>
          ))}
        </div>
           
      </div>
      
    </div>
    )
}



const heartButtonStyle = {
  height: "2rem",
  cursor: "pointer",
  color: "black",
}

const botStyle = {
  display: "flex",
  justifyContent: "space-between"
}

const titleStyle = {
  color: "#15827b"
}

const cardBorderStyle = {
  borderColor: "#6DCBC5",
  borderWidth: "3px",
  marginBottom: "0.5rem",
  marginLeft: "auto",
  marginRight: "auto",
  backgroundColor: "white"
}

const infoStyle = {
  color: "#6DCBC5"
}

const tagStyle = {
  color: "#368580",
  backgroundColor: "#deffe2",
  marginTop: "0.2rem", 
  marginBottom: "0.2rem", 

}

const imageStyle = {
}

export default ImageCard