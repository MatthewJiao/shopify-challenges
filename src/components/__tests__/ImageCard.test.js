import React from 'react'
import reactDom from 'react-dom'
import { cleanup, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ImageCard from '../ImageCard'

const image_test_data = {
    "id": 1509956,
    "pageURL": "https://pixabay.com/photos/woman-field-sunlight-fashion-hat-1509956/",
    "type": "photo",
    "tags": "woman, field, sunlight",
    "previewURL": "https://cdn.pixabay.com/photo/2016/07/11/15/43/woman-1509956_150.jpg",
    "previewWidth": 150,
    "previewHeight": 99,
    "webformatURL": "https://pixabay.com/get/g166b73dca4c700cd0a7cfd321cbf6e71447dfaa241a1e2ce5dbd38d67f1f05a5777ee02ddbc1921ad69748251c3c43c2435173a8de7bb63de5d51f77af154083_640.jpg",
    "webformatWidth": 640,
    "webformatHeight": 426,
    "largeImageURL": "https://pixabay.com/get/ge66b364c4be13d5c3ce681fd7dd3c45e5b21c607c7b374747fb447adcae34ebf6ec62b0387214262bb33f4d26d1f524c13dfc6572c4b583b36cad9efdf927c37_1280.jpg",
    "imageWidth": 5760,
    "imageHeight": 3840,
    "imageSize": 5088127,
    "views": 887498,
    "downloads": 461158,
    "collections": 2283,
    "likes": 2195,
    "comments": 337,
    "user_id": 334088,
    "user": "JillWellington",
    "userImageURL": "https://cdn.pixabay.com/user/2018/06/27/01-23-02-27_250x250.jpg"
};

afterEach(cleanup)
it("image card renders properly and without issue", () => {
    const div = document.createElement("div");
    reactDom.render(<ImageCard image = {image_test_data} />, div);
})


it("matches snapshot", () => {
    const tree = renderer.create(<ImageCard image = {image_test_data}/>).toJSON();
    expect(tree).toMatchSnapshot()
})
