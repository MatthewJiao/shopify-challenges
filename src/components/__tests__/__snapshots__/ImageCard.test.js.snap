// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="max-w-sm rounded-md overflow-hidden shadow-lg"
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#6DCBC5",
      "borderWidth": "3px",
      "marginBottom": "0.5rem",
      "marginLeft": "auto",
      "marginRight": "auto",
    }
  }
>
  <img
    alt="https://pixabay.com/get/ge66b364c4be13d5c3ce681fd7dd3c45e5b21c607c7b374747fb447adcae34ebf6ec62b0387214262bb33f4d26d1f524c13dfc6572c4b583b36cad9efdf927c37_1280.jpg"
    className="w-full"
    src="https://pixabay.com/get/g166b73dca4c700cd0a7cfd321cbf6e71447dfaa241a1e2ce5dbd38d67f1f05a5777ee02ddbc1921ad69748251c3c43c2435173a8de7bb63de5d51f77af154083_640.jpg"
    style={Object {}}
  />
  <div
    className="px-6 py-4"
  >
    <div
      className="font-bold text-xl mb-2"
      style={
        Object {
          "color": "#15827b",
        }
      }
    />
    <ul
      style={
        Object {
          "color": "#6DCBC5",
        }
      }
    >
      <li
        style={
          Object {
            "color": "#31948e",
          }
        }
      >
        <strong>
          View:
        </strong>
         
        887498
      </li>
      <li>
        <strong>
          Downloads:
        </strong>
         
        461158
      </li>
      <li>
        <strong>
          Likes:
        </strong>
         
        2195
      </li>
    </ul>
  </div>
  <div
    className="px-6 py-4 pt-1"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div>
      <span
        className="inline-block rounded-md px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        style={
          Object {
            "backgroundColor": "#deffe2",
            "color": "#368580",
            "marginBottom": "0.2rem",
            "marginTop": "0.2rem",
          }
        }
      >
        #
        woman,
      </span>
      <span
        className="inline-block rounded-md px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        style={
          Object {
            "backgroundColor": "#deffe2",
            "color": "#368580",
            "marginBottom": "0.2rem",
            "marginTop": "0.2rem",
          }
        }
      >
        #
        field,
      </span>
      <span
        className="inline-block rounded-md px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
        style={
          Object {
            "backgroundColor": "#deffe2",
            "color": "#368580",
            "marginBottom": "0.2rem",
            "marginTop": "0.2rem",
          }
        }
      >
        #
        sunlight
      </span>
    </div>
  </div>
</div>
`;
